app/main/posts/views/post-view-list.directive.js

```diff
@@ -97,18 +97,37 @@ function PostListController(
         PostEndpoint.query(postQuery).$promise.then(function (postsResponse) {
             // Add posts to full set of posts
             // @todo figure out if we can store these more efficiently
-            Array.prototype.push.apply($scope.posts, postsResponse.results);
 
+            // TODO/question: feels like this logic should belong somewhere else in the application? But where?
```

Comment:
I would like to put this somewhere else but I don't think I have entirely gotten how the filtering works so I don't know where? Any hints?